pkg/jwt/github/provider.go

```diff
@@ -15,35 +22,70 @@ func init() {
 // Provider abstracts the authentication for github
 type Provider struct {
 	provider.Verifier
+
+	teams []Team
 }
 
 // Build acts like the constructor for a provider
 func (gp *Provider) Build(config config.Credentials) provider.Provider {
 	client := NewClient()
+	teams := gp.teamConfigsToTeam(config.Github.Teams)
 
 	return &Provider{
 		Verifier: provider.NewVerifierBasket(
-			NewTeamVerifier(teamConfigsToTeam(config.Github.Teams), client),
+			NewTeamVerifier(teams, client),
 			NewOrganizationVerifier(config.Github.Organizations, client),
 		),
+		teams: teams,
 	}
 }
 
 // GetClaims returns a JWT Map Claim
 func (gp *Provider) GetClaims(httpClient *http.Client) (jwt.MapClaims, error) {
 	client := NewClient()
 
-	user, err := client.CurrentUser(httpClient)
-	if err != nil {
-		return nil, err
+	var (
+		wg            sync.WaitGroup
+		user          *github.User
+		usersOrgTeams OrganizationTeams
+		errs          []error
+	)
+
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		res, err := client.CurrentUser(httpClient)
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to get github users"))
+			return
+		}
+		user = res
+	}()
+
+	go func() {
+		defer wg.Done()
+		res, err := client.Teams(httpClient)
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "failed to get github teams"))
+			return
+		}
+		usersOrgTeams = res
+	}()
+
+	wg.Wait()
+
+	if len(errs) > 0 {
```

Comment:
`if err != nil`

Reply:
I think it's a bit harder to understand the code aggregating the result from 2 calls to a single variable, but not a big deal will change